packages/m2m-gateway/src/services/eserviceService.ts

```diff
@@ -77,5 +77,44 @@ export function eserviceServiceBuilder(clients: PagoPAInteropBeClients) {
 
       return toM2MGatewayApiEServiceDescriptor(descriptor);
     },
+    async getEServiceDescriptors(
+      eserviceId: EServiceId,
+      {
+        state,
+        offset,
+        limit,
+      }: {
+        state?: m2mGatewayApi.EServiceDescriptorState;
+        offset: number;
+        limit: number;
+      },
+      { headers, logger }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceDescriptors> {
+      logger.info(
+        `Retrieving eservice descriptors for eservice with id ${eserviceId} states ${state} offset ${offset} limit ${limit}`
+      );
+
+      const {
+        data: { descriptors },
+      } = await clients.catalogProcessClient.getEServiceById({
+        params: { eServiceId: eserviceId },
+        headers,
+      });
+
+      const filteredDescriptors = descriptors.filter((descriptor) =>
+        state ? descriptor.state === state : true
+      );
```

Comment:
We could completely skip the iteration if state query filter is not set
```suggestion
      const filteredDescriptors = state
        ? descriptors.filter((descriptor) => descriptor.state === state)
        : descriptors;
```